aspnet/tutorials/dotnet-watch.rst

```diff
@@ -0,0 +1,198 @@
+
+.. literalinclude:: dotnet-watch/sample/WebApp/Controllers/MathController.cs
+   :language: c#
+   :lines: 12-17
+   :emphasize-lines: 5
+   :dedent: 4
+
+Fix the code by replacing ``a + b`` with ``a * b``.
+
+Save the file. The console output will show messages similar to the ones below, indicating that ``dotnet watch`` detected a file change and restarted the application.
+
+
+.. code-block:: bash
+
+  [DotNetWatcher] info: File changed: /Users/user/dev/aspnet/Docs/aspnet/tutorials/dotnet-watch/sample/WebApp/Controllers/MathController.cs
+  [DotNetWatcher] info: Running dotnet with the following arguments: run
+  [DotNetWatcher] info: dotnet process id: 39940
+  Project WebApp (.NETCoreApp,Version=v1.0) will be compiled because inputs were modified
+  Compiling WebApp for .NETCoreApp,Version=v1.0
+  Compilation succeeded.
+    0 Warning(s)
+    0 Error(s)
+  Time elapsed 00:00:03.3312829
+  
+  Hosting environment: Production
+  Content root path: /Users/user/dev/aspnet/Docs/aspnet/tutorials/dotnet-watch/sample/WebApp
+  Now listening on: http://localhost:5000
+  Application started. Press Ctrl+C to shut down.
+
+
+
+Refresh or open the webpage pointing to ``http://localhost:5000/api/math/product?a=4&b=5`` and you should get the correct result (20).
+
+Running tests using ``dotnet watch``
+------------------------------------
+
+So far we only used ``dotnet watch`` to run our application. The file watcher can also be used to run other ``dotnet`` commands like ``test`` or ``publish``.
+
+1. Open the ``WebAppTests`` folder that already has ``dotnet watch`` in ``project.json``.
+2. Run ``dotnet watch test``.
+
+If you previously fixed the bug in the ``MathController`` then you'll see an output similar to the one below, otherwise you'll see a test failure:
+
+.. code-block:: bash
+
+  WebAppTests user$ dotnet watch test
+  [DotNetWatcher] info: Running dotnet with the following arguments: test
+  [DotNetWatcher] info: dotnet process id: 40193
+  Project WebApp (.NETCoreApp,Version=v1.0) was previously compiled. Skipping compilation.
+  Project WebAppTests (.NETCoreApp,Version=v1.0) was previously compiled. Skipping compilation.
+  xUnit.net .NET CLI test runner (64-bit .NET Core osx.10.11-x64)
+    Discovering: WebAppTests
+    Discovered:  WebAppTests
+    Starting:    WebAppTests
+    Finished:    WebAppTests
+  === TEST EXECUTION SUMMARY ===
+     WebAppTests  Total: 2, Errors: 0, Failed: 0, Skipped: 0, Time: 0.259s
+  SUMMARY: Total: 1 targets, Passed: 1, Failed: 0.
+  [DotNetWatcher] info: dotnet exit code: 0
+  [DotNetWatcher] info: Waiting for a file to change before restarting dotnet...
+
+In either case, once all the tests run, the watcher will indicate that it waits for a file to change before restarting ``dotnet test``.
+
+3. Open the controller file in ``WebApp/Controllers/MathController.cs`` and change some code. If you haven't fixed the product bug, do it now. Save the file.
```

Comment:
files are italic, not code
Open the controller file in *WebApp/Controllers/MathController.cs*
I probably missed other files.